Clean up ScrollerPairMac, and fix a bug with RTL scrollbars

smfr opened the pull request:
#### 478618b3a4aa237f81272cd2cbe27426323d897e
<pre>
Clean up ScrollerPairMac, and fix a bug with RTL scrollbars
<a href="https://bugs.webkit.org/show_bug.cgi?id=253871">https://bugs.webkit.org/show_bug.cgi?id=253871</a>
rdar://106682101

Reviewed by Aditya Keerthi.

A future patch may need to add locking around code that touches NSScrollerImp*, so it&apos;s helpful
if the NSScrollerImp* are hidden inside ScrollerPairMac/ScrollerMac, so do that.

Also do some other cleanup:
1. Replace ScrollerMac::Orientation with the existing ScrollbarOrientation enum
2. Move the calls to -setUsePresentationValue: and setPresentationValue: inside ScrollerPairMac
3. Have ScrollingTreeScrollingNodeDelegateMac::updateScrollbarPainters() use ScrollerPairMac::valuesForOrientation()
   and fix that code to use scroll offsets instead of positions, which fixes a bug where RTL scrollbars would sometimes
   flicker to the wrong positions.
4. Rename the &apos;doubleValue&apos; param to ScrollableArea::computeScrollbarValueAndOverhang() to scrollbarValue.

Canonical link: <a href="https://commits.webkit.org/261673@main">https://commits.webkit.org/261673@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e4b8701a76e55e2d1a8f5ddeec6a03645ae3a6b5

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/112469 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/21617 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/90/builds/1121 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/4248 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/121048 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/116535 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/22956 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/12770 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/86/builds/5388 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/118236 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/17049 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/100242 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/105525 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/99004 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/88/builds/792 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/46058 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/13966 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/89/builds/825 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/95082 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/14659 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/10201 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/19998 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/52829 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/8133 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/16479 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on Source/WebCore/page/scrolling/mac/ScrollerMac.h:
```diff
@@ -27,6 +27,7 @@
 #if PLATFORM(MAC)
 
+#include "ScrollTypes.h"
 #include <WebCore/FloatPoint.h>
```
Maybe we can forward declare `FloatPoint` as part of this cleanup?

smfr replied:
FloatPoint is referenced by value in the header.

pxlcoder replied:
I only see `WebCore::FloatPoint convertFromContent(const WebCore::FloatPoint&) const;` in this header, and we can forward declare return value types. Is there some place I'm missing?

Also, it's a little strange to see `WebCore::` in the type and `WebCore/` in the include, given that this file is in WebCore (an existing issue).

smfr committed:
Clean up ScrollerPairMac, and fix a bug with RTL scrollbars
https://bugs.webkit.org/show_bug.cgi?id=253871
rdar://106682101

Reviewed by Aditya Keerthi.

A future patch may need to add locking around code that touches NSScrollerImp*, so it's helpful
if the NSScrollerImp* are hidden inside ScrollerPairMac/ScrollerMac, so do that.

Also do some other cleanup:
1. Replace ScrollerMac::Orientation with the existing ScrollbarOrientation enum
2. Move the calls to -setUsePresentationValue: and setPresentationValue: inside ScrollerPairMac
3. Have ScrollingTreeScrollingNodeDelegateMac::updateScrollbarPainters() use ScrollerPairMac::valuesForOrientation()
   and fix that code to use scroll offsets instead of positions, which fixes a bug where RTL scrollbars would sometimes
   flicker to the wrong positions.
4. Rename the 'doubleValue' param to ScrollableArea::computeScrollbarValueAndOverhang() to scrollbarValue.

Canonical link: https://commits.webkit.org/261673@main
